### ISSUE TITLE
Save/load plugin state-preset (enhancement)

### ISSUE THREAD

**Issue by juanriccio** [apohl79/audiogridder#135]
It would be nice to be able to save and restore the preset of the audiogridded plugin. This should be doable on both the remote (server) and the local (client) side. Some of the advantages:

- much faster workflow when restarting the server: no need to hunt down the preset that was being used, or to tweak the parameters that had been tweaked in the UI.

- ease of backing up sessions or parts of sessions. Examples: when moving that plugin instance to a different server; when doing any kind of project recovery; when needing the same plugin in the same state in another project, etc.

- why both server and client side? So that backup preset files could be kept on two different machines. The savefile format could be different, depending on your implementation.

**Comment by apohl79** [comment]
I was always under the assumption, that you could use the preset functions of the DAW to store/restore a plugin chain. I never tried this though. I will have a look.

The server is designed to be stateless. All plugin settings live in your DAW. I'd like to keep the server slim in that regard, but extending the preset capabilities on the client site makes sense to me. Maybe presets could be managed on the client side, but persisted on the server, so other clients could load them? Or is that what you mean? Will think about it..

Thanks!

**Comment by juanriccio** [comment]
Also, I found it a bit hard to restore the state of a plugin. The client/server relationship gets flaky after a server restart - which happens frequently, since all it takes is changing some parameters or adding/removing plugins on a chain. Chains with multiple plugins are especially prone to fail. I can post more details about this when I get to it.

For now: what I experience is that most plugins loaded on the server lose their preset or settings, and no amount of saving/restoring with the DAW builtin system (Reaper in my experiments) will restore the session. The only reliable way I found is closing the project and reopening it. Hardly convenient.

That's why I was suggesting native save/restore from AudioGridder itself. It might be done from the plugin side only, of course, if the designer thinks it's the best solution.
